src/main/homework5/DZ1/Rectangle.java

```diff
@@ -0,0 +1,50 @@
+package main.homework5.DZ1;
+
+public class Rectangle extends Figure {
+
+    private int side1;
+    private int side2;
+
+
+    public Rectangle(int side1, int side2)
+    {
+
+        this.side1 = side1;
+        this.side2 = side2;
+    }
+
+
+    public int getSide1()
+    {
+        return side1;
+    }
+
+    public void setSide1(int side1)
+    {
+        this.side1 = side1;
+    }
+
+    public int getSide2()
```

Comment:
Since you've add setters for both side1 and side2, you've added a hack to break Square.
To avoid this, override one of these setters in the Square class to do nothing or better to update both sides